### PR TITLE
Moves inline JS from check.html to index.js

### PR DESCRIPTION
**Pull request by CliftonMcCallum**
Works on #4474

This PR migrated inline JS from check.html to index.js
### Technical

This can probably be placed in its own file, but I'm not certain how to do that with all of the different moving parts. I am open to suggestions. :)
### Testing

1. Go to http://localhost:8080/books/add
2. Add a test book that you plan to add a second time
3. Make certain the book was added by searching for it
4. Add the same book again
5. Make certain that you get a message like this with the redundant book info: "We already know about the 2020 testPublisher edition of testBook, but please, tell us more!"

### Screenshot

Evidence

### Stakeholders

@jdlrobson

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/index.js [hunk 1]
```diff
@@ -172,4 +172,17 @@ jQuery(function () {
     $('.column').disableSelection();
     $('.trash').disableSelection();
     $('#topNotice').hide();
+
+    // Functionality for check.html
+
+    $('.list-books a').click(function() {
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:178]: could you put this in a  file called add-book.js file which exports an init function

``
const addbook = require('./addbook.js');
addbook.init();
```

**Review by jdlrobson — CHANGES_REQUESTED** [review]

File: openlibrary/plugins/openlibrary/js/index.js (hunk 1 above)
Comment (CliftonMcCallum) [reply openlibrary/plugins/openlibrary/js/index.js:178]: Yes, no problem. Sorry that it took me a bit to get back to you. Thanks, @jdlrobson

**Comment by cclauss** [comment]
Labeling as blocked because of conflicting files and because our tests fail.

File: openlibrary/plugins/openlibrary/js/add-book.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export function initAddBookImport () {
+    window.q.push(function() {
+        $('.list-books a').click(function() {
+            var li = $(this).parents('li:first');
+            $('input#work').val('/works/' + $(li.attr('id')));
+            $('form#addbook').submit();
+        });
+
```
Comment (cclauss) [inline openlibrary/plugins/openlibrary/js/add-book.js:8]:
```suggestion

```

File: openlibrary/plugins/openlibrary/js/add-book.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+            $('input#work').val('/works/' + $(li.attr('id')));
+            $('form#addbook').submit();
+        });
+    
+        $('#bookAddCont').click(function() {
+            $('input#work').val('none-of-these');
+            $('form#addbook').submit();
+        });
+    });
+}
```
Comment (cclauss) [inline openlibrary/plugins/openlibrary/js/add-book.js:14]:
```suggestion
}

```

File: openlibrary/plugins/openlibrary/js/add-book.js (hunk 2 above)
Comment (CliftonMcCallum) [reply openlibrary/plugins/openlibrary/js/add-book.js:8]: Okay, that makes sense. I just deleted that space.

**Comment by cclauss** [comment]
https://eslint.org/docs/rules/prefer-template

**Comment by CliftonMcCallum** [comment]
Hey @jdlrobson , after reviewing the JS tests, I came across this on line 78:

- Jest: "global" coverage threshold for functions (19%) not met: 18.9%

Please let me know my next steps. I'm not sure what this means. Thanks!

**Comment by jdlrobson** [comment]
> Hey @jdlrobson , after reviewing the JS tests, I came across this on line 78:
> 
> * Jest: "global" coverage threshold for functions (19%) not met: 18.9%
> 
> Please let me know my next steps. I'm not sure what this means. Thanks!

It's asking you to write a unit test for the Javascript function you just changed. Are you up for the challenge either as part of this change or as a follow up?

File: openlibrary/plugins/openlibrary/js/add-book.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export function initAddBookImport () {
+    window.q.push(function() {
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/add-book.js:2]: No need to use window.q here. You can just jump into your code.

**Review by jdlrobson — CHANGES_REQUESTED** [review]
The window.q lines can be removed and the coverage issue needs to be fixed either via an edit to package.json to lower coverage (acceptable here so no pressure) or by providing a test (preferred)

https://github.com/internetarchive/openlibrary/blob/bcbd90a3b259fabb0bb79e8b8c234c25bf068d00/package.json#L72

**Review by CliftonMcCallum — COMMENTED** [review]
> the coverage issue needs to be fixed either via an edit to package.json to lower coverage (acceptable here so no pressure) or by providing a test (preferred)

I'll take a look at this today. Thank you for your help!

**Comment by CliftonMcCallum** [comment]
> > the coverage issue needs to be fixed either via an edit to package.json to lower coverage (acceptable here so no pressure) or by providing a test (preferred)
> 
> I'll take a look at this today. Thank you for your help!

Is there a coverage percentage that you think would be preferred for the package.json?

**Comment by jdlrobson** [comment]
The  window.q  issue here is the main problem. In terms of coverage you just need to increase the value to the minimum value required for the job to pass.

**Review by jdlrobson — APPROVED** [review]

**Pull request merged**